src/visit_vtk/full/vtkVisItCutter.C

```diff
@@ -1,21 +1,77 @@
-/*=========================================================================
+// Copyright (c) Lawrence Livermore National Security, LLC and other VisIt
+// Project developers.  See the top-level LICENSE file for dates and other
+// details.  No copyright assignment is required to contribute to VisIt.
 
-  Program:   Visualization Toolkit
-  Module:    $RCSfile: vtkVisItCutter.cxx,v $
-  Language:  C++
-  Date:      $Date: 2003/09/23 15:02:08 $
-  Version:   $Revision: 1.73 $
+#include "vtkVisItCutter.h"
 
-  Copyright (c) 1993-2002 Ken Martin, Will Schroeder, Bill Lorensen
-  All rights reserved.
-  See Copyright.txt or http://www.kitware.com/Copyright.htm for details.
+#if LIB_VERSION_GE(VTK,9,4,1)
 
-     This software is distributed WITHOUT ANY WARRANTY; without even
-     the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
-     PURPOSE.  See the above copyright notice for more information.
+#include <vtkInformation.h>
+#include <vtkInformationVector.h>
+#include <vtkUnstructuredGridBase.h>
 
-=========================================================================*/
-#include "vtkVisItCutter.h"
+
+vtkObjectFactoryNewMacro(vtkVisItCutter)
+
+//------------------------------------------------------------------------------
+vtkVisItCutter::vtkVisItCutter(vtkImplicitFunction* cf) : vtkCutter(cf)
+{
+  this->UnstructuredGridBypass = 0;
+}
+
+//------------------------------------------------------------------------------
+vtkVisItCutter::~vtkVisItCutter()
+{
+}
+
+// Cut through data generating surface.
+int vtkVisItCutter::RequestData(
+  vtkInformation* request,
+  vtkInformationVector** inputVector,
+  vtkInformationVector* outputVector)
+{
+  // get the info objects
+  vtkInformation* inInfo = inputVector[0]->GetInformationObject(0);
+  vtkInformation* outInfo = outputVector->GetInformationObject(0);
+
+  // get the input and output
+  vtkDataSet* input = vtkDataSet::SafeDownCast(inInfo->Get(vtkDataObject::DATA_OBJECT()));
+  vtkPolyData* output = vtkPolyData::SafeDownCast(outInfo->Get(vtkDataObject::DATA_OBJECT()));
+
+  if (!this->CutFunction)
+  {
+    vtkErrorMacro("No cut function specified");
+    return 0;
+  }
+
+  if (!input)
+  {
+    return 0;
+  }
+
+  if (input->GetNumberOfPoints() < 1 || this->GetNumberOfContours() < 1)
+  {
+    return 1;
+  }
+
+  if (this->UnstructuredGridBypass && vtkUnstructuredGridBase::SafeDownCast(input))
+  {
+      // bypasses the 'executePlaneCutter' for UnstructuredGrids
+      // Useful when Slicing polygons and polyhedra, as the executePlaneCutter
```

Comment:
I suspect this is not your commentary here but I am troubled with the verbiage here, "...useful when slicing polygons and polyhedra...". I don't think it is just "useful". Its necessary. And, that is due to lack of support (one might even say a bug) in `vtkPlaneCutter`.

Out of curiosity, do you have any idea why the solution from KitWare is not to simply fix the problem where it "garbbles cell data"? I am apt to believe that must be possible without noticeable loss of performance.

Reply:
@biagas...apologies...I misunderstood and I was being pendantic. I thought this code was cut-n-pasted from some other source in VTK proper.

Reply:
No worries @markcmiller86 . You made me realize that more comments are needed to explicitly state what is different from the base class.